backend/src/south/south-sqlite/south-sqlite.ts

```diff
@@ -56,10 +57,24 @@ export default class SouthSQLite extends SouthConnector implements QueriesHistor
     await super.start();
   }
 
-  // TODO: method needs to be implemented
   static async testConnection(settings: SouthConnectorDTO['settings'], logger: pino.Logger): Promise<void> {
     logger.trace(`Testing connection`);
-    throw new Error('TODO: method needs to be implemented');
+    const dbPath = path.resolve(settings.databasePath);
+    const dbFolder = path.dirname(dbPath);
+
+    try {
+      await fs.access(dbFolder, fs.constants.F_OK);
+    } catch (error: any) {
+      logger.trace(`Access error on '${dbFolder}': ${error.message}`);
+      throw new Error(`Folder '${dbFolder}' does not exist`);
+    }
+
+    try {
+      await fs.access(dbFolder, fs.constants.R_OK | fs.constants.W_OK);
+    } catch (error: any) {
+      logger.trace(`Access error on '${dbFolder}': ${error.message}`);
+      throw new Error(`No read/write access on folder`);
+    }
```

Comment:
I think we could improve this test a little bit more by querying the sqlite_master table : `SELECT * from sqlite_master` and log.info the result.

It should output something like on this screenshot and give useful info for debugging / testing / writing queries
<img width="730" alt="image" src="https://github.com/OptimistikSAS/OIBus/assets/28055217/08ec8711-944b-4510-8f92-8a93a74fd351">

By the way, we should query a system table on other sql connectors too, in order to test the connection properly

Reply:
Alright, and should we throw an error if the specified db file is not present? Because by default, the sqlite package creates the file if it's not present.

Reply:
Yes, let's first check if the file exists. If not, throw an error, if yes, request the database